tests/test_aird_filters.py

```diff
@@ -2,18 +2,96 @@
 # SPDX-License-Identifier: Apache-2.0
 
 """Tests for aird-filters applied after rendering a diagram"""
-import capellambse
+import pytest
+
+from capellambse import MelodyModel, aird
 
 COMP_PORT_FILTER_DIAG = "[LAB] Test Component Port Filter"
+EX_ITEMS_FILTER_DIAG = "[SDFB] Test ExchangeItem Filter"
+STD_ACTIVATED_FLTERS = [
+    "ModelExtensionFilter",
+    "hide.overlappedfunctional.chains.label.filter",
+    "hide.overlappedfunctional.chains.icon.filter",
+]
+EX_ITEMS_FLTER = "show.exchange.items.filter"
+NAME_AND_EX_ITEMS_FILTER = "show.functional.exchanges.exchange.items.filter"
 
 
-def test_component_ports_filter_is_applied(
-    model: capellambse.MelodyModel,
-) -> None:
+def test_diagram_has_activated_filters(model_5_2: MelodyModel) -> None:
+    diag = model_5_2.diagrams.by_name(EX_ITEMS_FILTER_DIAG)
+
+    assert diag.activated_filters == STD_ACTIVATED_FLTERS
+
+
+def test_add_activated_filter_to_diagram(model_5_2: MelodyModel) -> None:
+    diag = model_5_2.diagrams.by_name(EX_ITEMS_FILTER_DIAG)
+
+    diag.add_activated_filter(EX_ITEMS_FLTER)
+
+    assert diag.activated_filters == STD_ACTIVATED_FLTERS + [EX_ITEMS_FLTER]
+
+
+def test_remove_activated_filter_on_diagram(model_5_2: MelodyModel) -> None:
+    diag = model_5_2.diagrams.by_name(EX_ITEMS_FILTER_DIAG)
+
+    diag.remove_activated_filter(STD_ACTIVATED_FLTERS[0])
+
+    assert diag.activated_filters == STD_ACTIVATED_FLTERS[1:]
+
+
+def test_remove_activated_filter_fails(model_5_2: MelodyModel) -> None:
+    diag = model_5_2.diagrams.by_name(EX_ITEMS_FILTER_DIAG)
+
+    with pytest.raises(ValueError):
+        diag.remove_activated_filter(EX_ITEMS_FLTER)
+
+
+def test_component_ports_filter_is_applied(model: MelodyModel) -> None:
     diag = model.diagrams.by_name(COMP_PORT_FILTER_DIAG)
     diag = diag.render(None)
     assert all(
         d.hidden
         for d in diag
         if d.port and d.styleclass and d.styleclass.startswith("CP_")
     )
+
+
+def test_fex_exchangeitems_filter_is_applied(model_5_2: MelodyModel) -> None:
+    diag = model_5_2.diagrams.by_name(EX_ITEMS_FILTER_DIAG)
+    aird.RENDER_PARAMS["sorted_exchangedItems"] = False
+
+    diag.add_activated_filter(EX_ITEMS_FLTER)
+    diag = diag.render(None)
```

Comment:
My main point was that you were re-using a variable with a different type, which is only "valid" (i.e. mypy doesn't complain about it) because both assignments result in `typing.Any` and therefore are not type-checked at all.

The explicit annotations (to resolve `Any` to a concrete type) are a nice sprinkle on top, and probably a good idea generally when `Any` is involved. Although `typing.Any` will cripple type checks in every case, so they're still only of limited use.